offline/cable_cru_TRENDY.F90

```diff
@@ -436,65 +407,55 @@ SUBROUTINE cru_get_daily_met(CRU, LastDayOfYear)
   ! evil.
 
   ! Address prevTmax first
-  ! Check what the day is, and whether we need to change the year
+  ! Assume it's a regular day of the year
+  DummyYear = CRU%cYear
+  DummyDay = CRU%CTStep - 1
+
+  ! Special handling at the first day of the year
   IF (CRU%CTSTEP == 1) THEN
     ! Go back to previous year
     DummyYear = CRU%cYear - 1
-    IF (CRU%LeapYears) THEN
-      ! Check what the day should be
-      IF (is_leapyear(DummyYear)) THEN
-        DummyDay = 366
-      ELSE
-        DummyDay = 365
-      END IF
-    ELSE
-      ! No leapyears
-      DummyDay = 365
+    DummyDay = 365
+    IF ((CRU%LeapYears) .AND. (is_leapyear(DummyYear))) THEN
+      DummyDay = 366
     END IF
-  ELSE
-    ! Not the first day of the year, treat normally
-    DummyYear = CRU%cYear
-    DummyDay = CRU%CTStep - 1
   END IF
 
   ! Was the Tmax recycled?
-  IF (CRU%isRecycled(6)) THEN
+  IF (CRU%isRecycled(Tmax)) THEN
     DummyYear = 1901 + MOD(DummyYear - 1501, CRU%metRecyc)
-  ELSE
-    DummyYear = DummyYear
   END IF
 
   ! Now we just need to call cru_read_metvals with the Tmax Dataset reader and
   ! the prevTmax array to write to
-  CALL read_metvals(CRU%MetDatasets(6), CRU%Met(11)%MetVals, land_x, land_y,&
-    DummyYear, DummyDay, CRU%LeapYears, CRU%xDimSize, CRU%yDimSize,&
+  CALL read_metvals(CRU%MetDatasets(Tmax), CRU%Met(prevTmax)%MetVals, land_x,&
+    land_y, DummyYear, DummyDay, CRU%LeapYears, CRU%xDimSize, CRU%yDimSize,&
     CRU%DirectRead)
 
   ! Now do nextTmin
-  ! Check what the day is, and whether we need to change the year
-  ! This changes whether its a leapyear or not
-  IF ((CRU%LeapYears) .AND. (CRU%CTStep == 366)) THEN
-    ! We're at the last day of a leapyear
-    DummyDay = 1
-    DummyYear = CRU%cYear + 1
-  ELSEIF (CRU%CTStep == 365) THEN
-    ! We're at the end of a normal year
+  ! Assume it's a regular day of the year
+  DummyDay = CRU%CTStep + 1
+  DummyYear = CRU%cYear
+
+  ! Special handling at the last day of the year
+  DaysInYear = 365
+  IF (CRU%LeapYears) THEN
+    DaysInYear = 366
+  END IF
+
+  IF (CRU%CTStep == LastDayOfYear) THEN
```

Comment:
Was it what I had in the suggestion? Don't we need a `is_leapyear(CRU%cYear)` as well in the IF condition? Because we only have 366 days during a leap year, other years will have 365 days even if `CRU%LeapYears` is TRUE. Also you forgot to change `LastDayOfYear` to `DaysInYear`.

```suggestion
  DaysInYear = 365
  IF (CRU%LeapYears .AND. is_leapyear(CRU%cYear)) THEN
    DaysInYear = 366
  END IF

  IF (CRU%CTStep == DaysInYear) THEN
```